### PR TITLE
Added no sliders mod and pp calculator

### PR DESCRIPTION
**Pull request by Givikap120**
New no sliders mod that converts all sliders into notes. Added new convars that affect estimated pp value of a map in the main menu.

### CODE REVIEW

File: src/App/Osu/Osu.cpp [hunk 1]
```diff
@@ -65,7 +65,7 @@
 
 // release configuration
 bool Osu::autoUpdater = false;
-ConVar osu_version("osu_version", 33.05f);
+ConVar osu_version("osu_version", 33.06f);
```
Comment (McKay42) [inline src/App/Osu/Osu.cpp:68]: The version number is something only I update, not what you would include in a pr

File: src/App/Osu/OsuDatabaseBeatmap.cpp [hunk 2]
```diff
@@ -1900,12 +1908,26 @@ OsuDatabaseBeatmapStarCalculator::OsuDatabaseBeatmapStarCalculator() : Resource(
 	m_iMaxPossibleCombo = 0;
 }
 
+
+
+
+
+
```
Comment (McKay42) [inline src/App/Osu/OsuDatabaseBeatmap.cpp:1915]: What happened here? Don't commit random unrelated whitespace changes

File: src/App/Osu/OsuDatabaseBeatmap.cpp [hunk 3]
```diff
@@ -1900,12 +1908,26 @@ OsuDatabaseBeatmapStarCalculator::OsuDatabaseBeatmapStarCalculator() : Resource(
...
 void OsuDatabaseBeatmapStarCalculator::init()
 {
 	// NOTE: this accesses runtime mods, so must be run sync (not async)
 	// technically the getSelectedBeatmap() call here is a bit unsafe, since the beatmap could have changed already between async and sync, but in that case we recalculate immediately after anyways
 	if (!m_bDead.load() && m_iErrorCode == 0 && m_diff2->m_osu->getSelectedBeatmap() != NULL)
-		m_pp = OsuDifficultyCalculator::calculatePPv2(m_diff2->m_osu, m_diff2->m_osu->getSelectedBeatmap(), m_aimStars.load(), m_aimSliderFactor.load(), m_speedStars.load(), m_speedNotes.load(), m_iNumObjects.load(), m_iNumCircles.load(), m_iNumSliders.load(), m_iNumSpinners.load(), m_iMaxPossibleCombo);
+		if (osu_map_pp_calc_dynamic_acc_mode.getBool())
```
Comment (McKay42) [inline src/App/Osu/OsuDatabaseBeatmap.cpp:1921]: Not using { curly brackets } for a non-one-liner inside an if-statement is very bad style

File: src/App/Osu/OsuSongBrowser2.h [hunk 4]
```diff
@@ -255,6 +255,8 @@ class OsuSongBrowser2 : public OsuScreenBackable
 	void onModeChange(UString text);
 	void onModeChange2(UString text, int id = -1);
 
+	void onPpCalcChange(UString text);
```
Comment (McKay42) [inline src/App/Osu/OsuSongBrowser2.h:258]: "onPpCalcChange" is a bit extremely generic, make the function name more specific to your custom pp calc convars

File: src/App/Osu/OsuDatabaseBeatmap.cpp [hunk 5]
```diff
@@ -358,7 +366,7 @@ OsuDatabaseBeatmap::PRIMITIVE_CONTAINER OsuDatabaseBeatmap::loadPrimitiveObjects
 							colorOffset += (type >> 4) & 7; // special case 3: "Bits 4-6 (16, 32, 64) form a 3-bit number (0-7) that chooses how many combo colours to skip."
 						}
 
-						if (type & 0x1) // circle
+						if ((type & 0x1) || (osu_mod_no_sliders.getBool() && (type & 0x2))) // circle
```
Comment (McKay42) [inline src/App/Osu/OsuDatabaseBeatmap.cpp:369]: I know I suggested putting this logic inside loadPrimitiveObjects(), but actually this is a very bad idea since it will infect all pp calcs everywhere (meaning if a user recalculated their profile pp while this mod is enabled, all scores would suddenly be calculated as if having had that mod enabled).

Note: Experimental mods are _never_ considered for profile pp recalcs, that's why they are called "Experimental". However, having an experimental mod enabled should not affect unrelated calculations in other parts of the code as drastically as this one.

You want to have correct pp values if sliders are turned into circles, but it shouldn't affect _everything_ globally. This is problematic because there is no infrastructure to support that logic currently. As mentioned, I don't care for pp/star accuracy when experimental mods are used, which is why this is the way it is. Therefore this can't be merged as is.

Comment (McKay42) [reply src/App/Osu/OsuDatabaseBeatmap.cpp:369]: My suggestion would be to ignore accurate pp/stars and put the if-condition inside loadGameplay() instead.

File: src/App/Osu/OsuDatabaseBeatmap.cpp (hunk 3 above)
Comment (Givikap120) [reply src/App/Osu/OsuDatabaseBeatmap.cpp:1921]: I didn't even notice that there are no {}
